x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1744,23 +1733,34 @@ AtomicLong getLastEvictionCheckedAt() {
      */
     public static String getCreatorRealmName(final Authentication authentication) {
         if (authentication.isApiKey()) {
-            return (String) authentication.getMetadata().get(AuthenticationField.API_KEY_CREATOR_REALM_NAME);
+            return (String) authentication.getAuthenticatingSubject().getMetadata().get(AuthenticationField.API_KEY_CREATOR_REALM_NAME);
         } else {
             return authentication.getSourceRealm().getName();
```

Comment:
I wanted to replace this as well, but `getCreatorRealmName` gets used in audit logging and working out the ramifications of either throwing/returning `null` got a little hairy so I'm leaving this for another day.

Reply:
Perhaps useful to leave a comment here unless you plan to work on it right away.